LSPS5/README.md

```diff
@@ -0,0 +1,658 @@
+    with the notification delivery service and then getting a
+    token in return, and only putting the token in the URI.
+* MAY encode any additional information in the URI.
+
+> **Rationale** By placing most data in the URI, the LSP
+> interface is simplified and the client developer has the
+> flexibility to add any additional information they want
+> the LSP to simply pass through from the client instance
+> to the notification delivery service.
+
+If a webhook with the given `app_name` is already registered,
+then this call simply replaces the existing webhook and
+succeeds.
+Otherwise, the LSP MUST check if the number of registered
+webhooks is already at the limit it imposes.
+If it is still below the limit, the LSP inserts the new
+webhook under the given `app_name` and succeeds.
+(i.e. this is an "insert-or-replace" operation with a maximum
+number of inserted webhooks)
+
+The LSP:
+
+* MUST persist the given webhook and associate it with the client
+  node ID and given `app_name` before responding to this call.
+* MUST use **all** `webhook`s that were registered by successful
+  `lsps5.set_webhook` calls, ***except*** for the
+  `lsps5.webhook_registered` notification.
+* MUST remember all webhooks:
+  * If there are currently no channels with this client, for at
+    least 7 days.
+    * If a new channel with the client is opened, then as below.
+  * If there is at least one open channel with this client,
+    indefinitely remember them, until for at least 7 days after
+    the last channel with the client is closed.
+
+On success, `lsps5.set_webhook` returns an object like the
+following:
+
+```JSON
+{
+  "num_webhooks": 2,
+  "max_webhooks": 4,
+  "no_change": false
+}
+```
+
+`num_webhooks` is the number of webhooks already registered,
+including this one if it added a new webhook.
+`max_webhooks` is the maximum number of webhooks the LSP allows
+per client.
+Both are unsigned JSON integers.
+
+`no_change` is a Boolean.
+If `true`, then the client has used `lsps5.set_webhook` before,
+with the exact same `app_name` and `webhook` as the LSP has in
+its storage.
+If `false`, then either the `app_name` is new, or the
+corresponding `webhook` was changed.
+
+If `no_change` is `false`, then the LSP will send the
+`lsps5.webhook_registered` notification to the `webhook`
+specified in this call, and only to that webhook.
+Only the newly-registered webhook is contacted for this
+notification (unlike other events, which contact all registered
+webhooks).
+The LSP MAY send this notification before or after responding to
+the `lsps5.set_webhook` call.
+The LSP MUST send this notification to this webhook before sending
+any other notifications to this webhook.
+
+The `lsps5.set_webhook` call has the following errors defined
+(error `code` number in parenthesis):
+
+* `too_long` (1000) - Either `app_name` or `webhook` or both are
+  too long.
+* `unsupported_protocol` (1001) - The `webhook` is not a protocol
```

Comment:
should there also be an error for invalid webhook url? if it doesn't parse/validate as a RFC 1738 url?

Reply:
Add  `url_parse_error` code.